src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -452,6 +458,18 @@ public IReadOnlyList<TResult> InvokePSCommand<TResult>(PSCommand psCommand, Powe
             return task.ExecuteAndGetResult(cancellationToken);
         }
 
+        void IInternalPowerShellExecutionService.UnsafeInvokePSCommand(
```

Comment:
**UNSAFE**

Reply:
> **UNSAFE**

Yeah it's a convention used in a few places in the BCL to just mean "make sure you know what you're doing".

Mainly I don't like that we have `ExecuteCommandAsync` which marshals to the right thread... and then we have `InvokeCommand` which assumes you're on the right thread and potentially dangerously just invokes under that assumption. 

It would be very easy to mess up our threading house of cards by forgetting this and assuming `InvokeCommand` is just "the sync version that does the same thing".